tests/unittests/Foundation/NSFileManagerTests.mm

```diff
@@ -136,4 +170,76 @@
     NSString* currentPath = [manager currentDirectoryPath];
     ASSERT_OBJCNE_MSG(originalPath, currentPath, "Expected change in current directory");
     ASSERT_OBJCEQ_MSG(parentPath, currentPath, "Expected current directory to change to parentPath");
-}
\ No newline at end of file
+}
+
+TEST(NSFileManager, MoveFileViaPath) {
+    char const* srcName = "\\NSFileManagerMoveTestFile.txt";
+    char const* destName = "\\MovedFile.txt";
+
+    NSString* content = @"The Quick Brown Fox.";
+    createFileWithContentAndVerify(srcName, content);
+
+    NSString* srcPath = getPathToFile(srcName);
+    NSString* destPath = getPathToFile(destName);
+
+    NSFileManager* manager = [NSFileManager defaultManager];
+
+    NSError* error = nil;
+    BOOL status = [manager moveItemAtPath:srcPath toPath:destPath error:&error];
+    EXPECT_TRUE(status);
+    EXPECT_EQ(nil, error);
+
+    // Verify file exists.
+    EXPECT_TRUE([manager fileExistsAtPath:destPath]);
+    EXPECT_FALSE([manager fileExistsAtPath:srcPath]);
+
+    // Verify data.
+    NSFileHandle* fh = [NSFileHandle fileHandleForReadingAtPath:destPath];
+    EXPECT_TRUE(fh != nil);
+
+    NSData* fileData = [fh readDataToEndOfFile];
+    EXPECT_TRUE_MSG(fileData != nil, "FAILED: reading the entire file failed.");
+
+    NSString* str = [[[NSString alloc] initWithData:fileData encoding:NSUTF8StringEncoding] autorelease];
+    EXPECT_OBJCEQ_MSG(str, content, "FAILED: Unable to read the file content.");
+
+    [fh closeFile];
+
+    deleteFile((status ? destName : srcName));
+}
+
+TEST(NSFileManager, MoveFileViaURL) {
+    char const* srcName = "\\NSFileManagerMoveTestFile.txt";
+    char const* destName = "\\MovedFile.txt";
+
+    NSString* content = @"The Quick Brown Fox.";
+    createFileWithContentAndVerify(srcName, content);
+
+    NSURL* srcURL = [NSURL fileURLWithPath:getPathToFile(srcName)];
+    NSURL* destURL = [NSURL fileURLWithPath:getPathToFile(destName)];
+
+    NSFileManager* manager = [NSFileManager defaultManager];
+
+    NSError* error = nil;
+    BOOL status = [manager moveItemAtURL:srcURL toURL:destURL error:&error];
+    EXPECT_TRUE(status);
+    EXPECT_EQ(nil, error);
+
+    // Verify file exists.
+    EXPECT_TRUE([manager fileExistsAtPath:[destURL path]]);
+    EXPECT_FALSE([manager fileExistsAtPath:[srcURL path]);
+
+    // Verify data.
+    NSFileHandle* fh = [NSFileHandle fileHandleForReadingAtPath:getPathToFile(destName)];
+    EXPECT_TRUE(fh != nil);
+
+    NSData* fileData = [fh readDataToEndOfFile];
+    EXPECT_TRUE_MSG(fileData != nil, "FAILED: reading the entire file failed.");
+
+    NSString* str = [[[NSString alloc] initWithData:fileData encoding:NSUTF8StringEncoding] autorelease];
```

Comment:
This is not testing whether we can read UTF-8, and additional encodings are out-of-scope here.
It'd probably be better to avoid using NSString entirely for this and just compare two NSData instances.

---
In reply to: [71408245](https://github.com/Microsoft/WinObjC/pull/669#discussion_r71408245) [](ancestors = 71408245)